lib/src/merge.rs

```diff
@@ -806,6 +833,88 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_deduplicate() {
+        // 1-way merge
+        assert_eq!(c(&[], &[0]).deduplicate(), c(&[], &[0]));
+        // 3-way merge
+        assert_eq!(c(&[0], &[0, 0]).deduplicate(), c(&[0], &[0, 0]));
+        assert_eq!(c(&[0], &[0, 1]).deduplicate(), c(&[0], &[0, 1]));
+        assert_eq!(c(&[0], &[1, 0]).deduplicate(), c(&[0], &[1, 0]));
+        assert_eq!(c(&[0], &[1, 1]).deduplicate(), c(&[0], &[1, 1]));
+        assert_eq!(c(&[0], &[1, 2]).deduplicate(), c(&[0], &[1, 2]));
+        // 5-way merge
+        assert_eq!(c(&[0, 0], &[0, 0, 0]).deduplicate(), c(&[0], &[0, 0]));
+        assert_eq!(c(&[0, 0], &[0, 0, 1]).deduplicate(), c(&[0, 0], &[0, 0, 1]));
```

Comment:
I feel like this one should have been deduplicated to `c(&[0], &[0, 1])`. Similarly, the next line seems like it should be `c(&[0], &[0, 1])`. And even things like `c(&[0, 1, 2, 3, 0], &[4, 5, 6, 6, 7, 8])` should lose the duplicate `0->6` diffs and become `c(&[0, 1, 2, 3], &[4, 5, 6, 8, 7])`. Note how I tried to keep the diffs as well as possible by letting the `8` take the dropped `6`'s place so the `3->7` diff remains. I think that matches what `simplify()` does.

Reply:
The current implementation is just a very naive search for exact duplicated remove-add pairs. Am I correct to say that any duplicated pair of adds and removes can be removed from the conflict list irregardless of whether the add/remove are contiguous?

Reply:
Exactly.

Reply:
Updated to do this. The code isn't the most efficient (there's possibly a better algorithm somewhere), but I don't think that's a huge issue since most `Merge`s won't have a large number of values anyway.